src/cascadia/TerminalControl/TermControl.cpp

```diff
@@ -619,9 +619,24 @@ namespace winrt::Microsoft::Terminal::TerminalControl::implementation
         }
 
         const auto modifiers = _GetPressedModifierKeys();
-        const auto vkey = static_cast<WORD>(e.OriginalKey());
 
+        // AltGr key combinations don't always contain any meaningful,
+        // pretranslated unicode character during WM_KEYDOWN.
+        // E.g. on a German keyboard AltGr+Q should result in a "@" character,
+        // but actually results in "Q" with Alt and Ctrl modifier states.
+        // By returning false though, we can abort handling this WM_KEYDOWN
+        // event and let the WM_CHAR handler kick in, which will be
+        // provided with an appropriate unicode character.
+        if (modifiers.IsAltGrPressed())
+        {
+            _SendKeyEvent();

```

Comment:
I’m not sure why we need to send an empty key event here.

Reply:
The key event handler must call `_terminal->ClearSelection()`, because the char handler doesn't.
Additionally `_terminal->SetCursorVisible(true);` must be called on a key press as the comment at line 683 states.